library/Rules/Email.php

```diff
@@ -1,10 +1,32 @@
 <?php
 namespace Respect\Validation\Rules;
 
+use Egulias\EmailValidator\EmailValidator;
+
 class Email extends AbstractRule
 {
+    public function __construct(EmailValidator $emailValidator = null)
+    {
+        $this->emailValidator = $emailValidator;
+    }
+
+    public function getEmailValidator()
+    {
+        if (!$this->emailValidator instanceof EmailValidator
```

Comment:
Anyone can change any property of an object, even the private ones (using the reflection API), doing this we avoid other problems (it's kinda neurosis, I know).